components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/dao/DeviceFlowDAOImpl.java

```diff
@@ -103,16 +144,66 @@ public void setAuthenticationStatus(String userCode, String status) throws Ident
                 IdentityDatabaseUtil.commitTransaction(connection);
             } catch (SQLException e) {
                 IdentityDatabaseUtil.rollbackTransaction(connection);
-                throw new IdentityOAuth2Exception("Error when setting user has authenticated for user_code: " +
-                                                  userCode, e);
+                throw new IdentityOAuth2Exception("Error when setting the authentication status for the user_code: " +
+                        userCode, e);
             }
         } catch (SQLException e) {
-            throw new IdentityOAuth2Exception("Error when setting user has authenticated for user_code: " + userCode,
-                                              e);
+            throw new IdentityOAuth2Exception("Error when setting the authentication status for the user_code: " +
+                    userCode, e);
         }
     }
 
     @Override
+    public DeviceFlowDO getAuthenticationDetails(String deviceCode, String clientId) throws IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Getting authentication details for device_code: " + deviceCode);
+        }
+        AuthenticatedUser user;
+        int tenantId = 0;
+        String userName = null;
+        boolean checked = false; // Check for valid deviceCode and clientId.
+        String userDomain = null;
+        String authenticatedIDP = null;
+        DeviceFlowDO deviceFlowDO = new DeviceFlowDO();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement prepStmt =
+                     connection.prepareStatement(SQLQueries.DeviceFlowDAOSQLQueries.GET_AUTHENTICATION_STATUS)) {
+            prepStmt.setString(1, deviceCode);
+            prepStmt.setString(2, clientId);
+            try (ResultSet resultSet = prepStmt.executeQuery()) {
+                while (resultSet.next()) {
+                    deviceFlowDO.setStatus(resultSet.getString(1));
+                    deviceFlowDO.setLastPollTime(resultSet.getTimestamp(2,
+                            Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))));
+                    deviceFlowDO.setPollTime(resultSet.getLong(3));
+                    deviceFlowDO.setExpiryTime(resultSet.getTimestamp(4,
+                            Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))));
+                    userName = resultSet.getString(5);
+                    tenantId = resultSet.getInt(6);
+                    userDomain = resultSet.getString(7);
+                    authenticatedIDP = resultSet.getString(8);
+                    checked = true;
```

Comment:
Shall we rename this to something more meaningful

Reply:
fixed